contracts/utils/RelayHubValidator.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier:MIT
+pragma solidity ^0.7.5;
+pragma abicoder v2;
+
+import "../interfaces/GsnTypes.sol";
+library RelayHubValidator {
+
+    // validate that encoded relayCall is properly packed without any extra bytes
+    function verifyTransactionPacking(
+        GsnTypes.RelayRequest calldata relayRequest,
+        bytes calldata signature,
+        bytes calldata approvalData
+    ) internal pure {
+        uint expectedMsgDataLen = 4 + 22*32 +
```

Comment:
please use explicit integer sizes

Reply:
the old elaborated description was lost. added back.